### PR TITLE
Check first if property type is Guid and if so then handle it separat…

### PR DESCRIPTION
**Pull request by cocojiambo**
Fix for Issue #194

### CODE REVIEW

File: test/Carter.Tests/Modelbinding/TestModel.cs [hunk 1]
```diff
@@ -15,5 +16,7 @@ public class TestModel
         public IEnumerable<int> MyIntArrayProperty { get; set; }
 
         public IEnumerable<int> MyIntListProperty { get; set; }
+
+        public Guid MyGuidListProperty { get; set; }
```
Comment (jchannon) [inline test/Carter.Tests/Modelbinding/TestModel.cs]: Can you rename this to `MyGuidProperty` please?

File: src/ModelBinding/BindExtensions.cs [hunk 2]
```diff
@@ -62,6 +62,17 @@ public static T Bind<T>(this HttpRequest request)
                         }
                         return val.Value.Select(y => Convert.ChangeType(y, colType));
                     }
+
+                    // Convert.ChangeType (below) chokes on Guids, so doing it here instead.
+                    if (propertyType == typeof(Guid))
```
Comment (jchannon) [inline src/ModelBinding/BindExtensions.cs]: will this work if it is `Guid?`

File: test/Carter.Tests/Modelbinding/TestModel.cs (hunk 1 above)
Comment (cocojiambo) [reply test/Carter.Tests/Modelbinding/TestModel.cs]: Yeap! Will do.

File: src/ModelBinding/BindExtensions.cs (hunk 2 above)
Comment (cocojiambo) [reply src/ModelBinding/BindExtensions.cs]: Good question! Let me come back to you.

Comment (cocojiambo) [reply src/ModelBinding/BindExtensions.cs]: Ok, I have pushed the changes now. Are you able to see them?

File: test/Carter.Tests/Modelbinding/TestModel.cs (hunk 1 above)
Comment (cocojiambo) [reply test/Carter.Tests/Modelbinding/TestModel.cs]: Done.

File: src/ModelBinding/BindExtensions.cs (hunk 2 above)
Comment (cocojiambo) [reply src/ModelBinding/BindExtensions.cs]: Ok, I can see that the changes are here. Let me know if you find anything else wrong with it. I did a test with both `Guid` and `Guid?` and it worked fine now.

**Comment by jchannon** [comment]
Looks good, only thing I'm unsure of is the sln changes. I'll pull it down later to see if I can open it still :)

**Comment by cocojiambo** [comment]
I would ignore the `sln` changes and pull in only the code changes. Are you able to cherry-pick?

**Comment by jchannon** [comment]
Afraid not

**Comment by cocojiambo** [comment]
Shall I delete it from the PR?

**Comment by jchannon** [comment]
yes please

**Comment by jchannon** [comment]
this looks like you've deleted the sln file.

you'll need to checkout the sln file from the master branch so there are no differences

`git checkout master -- Carter.sln`

**Comment by cocojiambo** [comment]
Its my first ever PR in GitHub so I am trying to find my way round it. Where do I run that command please?

**Comment by jchannon** [comment]
No worries.

You would run that in a terminal on your machine

**Comment by cocojiambo** [comment]
Let me do it again, because its gone very wrong.

**Pull request closed without merging**